client/starwhale/api/_impl/argument.py

```diff
@@ -0,0 +1,216 @@
+from __future__ import annotations
+
+import typing as t
+import inspect
+import threading
+import dataclasses
+from enum import Enum
+from functools import wraps
+
+import click
+
+from starwhale.utils import console
+
+
+# TODO: use a more elegant way to pass extra cli args
+class ExtraCliArgsRegistry:
+    _args = None
+    _lock = threading.Lock()
+
+    @classmethod
+    def set(cls, args: t.List[str]) -> None:
+        with cls._lock:
```

Comment:
is there any multiple threading cases? what if a following thread modifies the args

Reply:
Not Sure. Only keep `ExtraCliArgsRegistry` thread safe.